src/core/gltf2importer/gltf2parser.cpp

```diff
@@ -913,50 +914,70 @@ void GLTF2Parser::generateTreeNodeContent()
                         // TO DO: generate proper morph target vertex shader based
                         // on meshData.morphTargetCount and primitiveData.morphTargets
 
+                        auto setBrdfLutOnEffect = [this](GLTF2Material *m) {
+                            auto effect = qobject_cast<MetallicRoughnessEffect *>(m->effect());
+                            if (effect && m_sceneEntity) {
+                                auto texture = m_sceneEntity->texture(QLatin1String("_kuesa_brdfLUT"));
+                                effect->setBrdfLUT(texture);
+                            }
+                        };
+
+                        auto checkMaterialIsCompatibleWithPrimitive = [](GLTF2Material *m, Qt3DRender::QGeometryRenderer *renderer,
+                                                                         const QString &primitiveName, const QString &materialName) {
+                            MetallicRoughnessMaterial *metalRoughMat = qobject_cast<MetallicRoughnessMaterial *>(m);
+                            // When we have a normal map on a MetalRoughMetarial, make sure we have a tangent attribute
+                            if (metalRoughMat != nullptr && metalRoughMat->normalMap() != nullptr) {
+                                Q_ASSERT(renderer->geometry());
+                                const QVector<Qt3DRender::QAttribute *> attributes = renderer->geometry()->attributes();
+                                bool hasTangentAttribute = std::find_if(attributes.cbegin(),
+                                                                        attributes.cend(),
+                                                                        [](Qt3DRender::QAttribute *attr) {
+                                                                            return attr->name() == Qt3DRender::QAttribute::defaultTangentAttributeName();
+                                                                        }) != attributes.cend();
+                                if (!hasTangentAttribute)
+                                    qWarning() << QStringLiteral("Primitive %1 is trying to use Material %2 which does normal mapping even though it defines no tangent attribute. This can result in incorrect rendering, please consider generating tangents.")
```

Comment:
this is going to be weird if the material or primitive doesn't have a name. Ideally could use index but I don't think we have access to that info easily at this point...

Reply:
I could point out the primitive and/or material are nameless if that can be better